inc/endpoints/class-components-endpoint.php

```diff
@@ -359,7 +361,9 @@ public function build_query() {
 		// Execute query.
 		$wp_query = new \WP_Query( $query );
 
-		if ( empty( $wp_query->posts ) && ! $wp_query->is_search() ) {
+		if ( '/' === $this->path ) {
```

Comment:
This is not technically correct, because if a site is set to have another page as the page for posts, _that_ page should return true `is_home()` and the root path should not. Instead, the root path should return `is_front_page()`. 

This is tricky because there is some logic in `WP_Query::parse_query()` which sets the `is_home` flag based on a bunch of conditionals—one of which is (annoyingly) that this isn't a REST request. [See the source](https://github.com/WordPress/WordPress/blob/5.3/wp-includes/class-wp-query.php#L960-L962).

Afterward, this flag is used in the logic for several other conditionals, so we either need to manually fix that check when we build the query and then parse the query again, or we need to fork all of the parse query logic that depends on the `is_home` flag. Either way, I think that's outside the scope of this PR so I'm going to log a new issue for it and we can fix in a separate issue.

Reply:
Sounds good - yeah, this was solely meant to be a quick hotfix for the sake of testing other functionality. I agree it needs more discussion and consideration before a full solution can be developed.

Reply:
Opened #139 for this.

Reply:
Do you think this is a fine temporary solution? Or should we remove the "fix" from this PR? I'm okay either way.